File: java-playground/src/test/java/org/gonza/javaplayground/util/RandomNumberGeneratorTest.java
package org.gonza.javaplayground.util;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.List;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class RandomNumberGeneratorTest {
	@Test
	@DisplayName("사이즈를 3으로 정의하면 3자리이다")
	void size() {
		List<Integer> randomNumbers = RandomNumberGenerator.generate(3, 1, 9);
		assertThat(randomNumbers.size()).isEqualTo(3);
	}

	@Test
	@DisplayName("1에서 9 사이의 숫자가 생성된다")
	void range() {
		List<Integer> randomNumbers = RandomNumberGenerator.generate(3, 1, 9);
		randomNumbers.forEach(number -> {
			assertThat(number).isGreaterThan(0);
			assertThat(number).isLessThan(10);
		});
	}
}

File: kotlin-playground/src/test/kotlin/org/gonza/kotlinplayground/RandomNumberGeneratorTest.kt
package org.gonza.kotlinplayground

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class RandomNumberGeneratorTest {
    @Test
    fun `랜덤 번호는 0-9 사이의 값이다`() {
        val randomNumberGenerator = RandomNumberGenerator(10)
        val generatedNumber = randomNumberGenerator.generate()

        val expectedValue = (0..9).toList()

        Assertions.assertThat(expectedValue).contains(generatedNumber)
    }
}
